lib/puppet/ssl/certificate_authority.rb

```diff
@@ -288,14 +288,15 @@ def sign(hostname, options={})
       cert_type = :ca
       issuer = csr.content
     else
-      options[:allow_dns_alt_names] = true if hostname == Puppet[:certname].downcase
       unless csr = Puppet::SSL::CertificateRequest.indirection.find(hostname)
         raise ArgumentError, "Could not find certificate request for #{hostname}"
       end
 
       cert_type = :server
       issuer = host.certificate.content
 
+      # This allows for various bootstrapping scenarios
```

Comment:
So this change moves what was being set at line 291 down to where it's actually consumed and comments why we do this overriding of default behavior in the first place. This comment and line of code are then copied to the [certificate authority interface's sign method](https://github.com/puppetlabs/puppet/pull/5120/files/1a9a2f7217f5b3d3dfdfbaa6dc4a369ba3f17aab#diff-ebefe776564cbe300da0fb3a405e0713R283) where we now also call `check_internal_signing_policies`.

That copy pasta train wreck is the "fix" because we now call `check_internal_signing_policies` before we call `sign` when using the `puppet cert` command (so we can get potential issues to the user faster). The bug was that the conditional overriding of default behavior didn't follow the method invocation into the certificate authority interface (we saw this with our duplicate cert acceptance test).

Thinking about this more, I think it might be a better idea to move the hostname == certname check into the `check_internal_signing_policies` so that the callers don't have to be concerned (ie this check seems to be part of our internal signing policies). What do you guys think?